practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/client/RoleServiceClient.java

```diff
@@ -1,18 +1,17 @@
 package uk.ac.ox.ndph.mts.practitioner_service.client;
 
-import uk.ac.ox.ndph.mts.practitioner_service.exception.RestException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
 
-/**
- * Interface for client to the REST role-service
- */
-public interface RoleServiceClient {
+@Component
+public class RoleServiceClient extends AbstractEntityServiceClient {
 
-    /**
-     * Check if a role ID exists
-     * @param id ID to check
-     * @return true if exists, false if not
-     * @throws RestException on communication error only
-     */
-    boolean roleIdExists(String id) throws RestException;
+    public RoleServiceClient(final WebClient.Builder webClientBuilder,
+                             @Value("${role.service.uri}") String serviceUrlBase) {
+        this.serviceUrlBase =  serviceUrlBase;
+        this.webClient = webClientBuilder.baseUrl(serviceUrlBase).build();
+        this.serviceExistsRoute = "/roles/{id}";
+    }
```

Comment:
Initializing protected members in the derived class constructor - shouldn't this be done in the base class constructor? Risk of members being not properly initialized.